data/tutorials/lg_00_data_types_and_matching.md

```diff
@@ -545,19 +550,11 @@ what are known as *guards* to each pattern match. A guard is the conditional
 which follows the `when`, and it means that the pattern match only happens if
 the pattern matches *and* the condition in the `when`-clause is satisfied.
 
-<!-- $MDX skip -->
-```ocaml
-match value with
-| pattern [ when condition ] -> result
-| pattern [ when condition ] -> result
-  ...
-```
-
-The second feature is the `=` operator which tests for "structural equality"
-between two expressions. That means it goes recursively into each expression
-checking they're exactly the same at all levels down.
+The second feature is the `=` operator, which tests for "structural equality"
+between two expressions. That means it goes recursively into each expression,
+checking they're exactly the same at all levels.
 
-Another feature which is useful when we build more complicated nested patterns
+Another useful feature when building more complicated nested patterns
 is the `as` keyword, which can be used to name part of an expression. For
 example:
 
```

Comment:
I would change

<!-- $MDX skip -->
```ocaml
Name ("/DeviceGray" | "/DeviceRGB" | "/DeviceCMYK") as n -> n

Node (l, ((k, _) as pair), r) when k = k' -> Some pair
```

to

<!-- $MDX skip -->
```ocaml
| Name ("/DeviceGray" | "/DeviceRGB" | "/DeviceCMYK") as n -> n

| Node (l, ((k, _) as pair), r) when k = k' -> Some pair
```

as there is a more clear visual reference to these being part of a pattern match.

In this instance, it is fine that it is not directly copyable to a REPL, I think.

However, we need to figure out a good way to reliably link to exercises from within the tutorials, or to have links to pre-filled playgrounds that illustrate the concept in such a way that the reader can tinker with the code. These examples focus on something very specific that isn't fully defined in the text.

Reply:
I changed that as suggested. I like that idea of linking to something where they can play with the code!